src/async-snowplow.test.ts

```diff
@@ -0,0 +1,186 @@
+import { ImmediateAsyncSnowplow, TimerAsyncSnowplow } from './async-snowplow';
+
+jest.useFakeTimers();
+
+describe('ImmediateAsyncSnowplow', () => {
+  describe('call', () => {
+    it('success', () => {
+      const snowplow = jest.fn();
+      const asyncSnowplow = new ImmediateAsyncSnowplow(snowplow, (e) => {
+        throw e;
+      });
+      const event = { fake: 'prop' };
+      asyncSnowplow.call('trackUnstructEvent', event);
+      expect(snowplow.mock.calls.length).toBe(1);
+      expect(snowplow.mock.calls[0][0]).toEqual('trackUnstructEvent');
+      expect(snowplow.mock.calls[0][1]).toEqual(event);
+    });
+
+    it('success', () => {
+      const snowplow = jest.fn(() => {
+        throw 'Failed';
+      });
+      const asyncSnowplow = new ImmediateAsyncSnowplow(snowplow, (e) => {
+        throw e;
+      });
+      const event = { fake: 'prop' };
+      expect(() => asyncSnowplow.call('trackUnstructEvent', event)).toThrow(/^Failed$/);
+    });
+  });
+
+  it('tryFlush', () => {
+    const snowplow = jest.fn();
+    const asyncSnowplow = new ImmediateAsyncSnowplow(snowplow, (e) => {
+      throw e;
+    });
+    asyncSnowplow.tryFlush();
+    // Nothing should happen.
```

Comment:
Is this something you could check via assertions?

Reply:
Hmm, I don't have a great way to check side effects.  I mostly want it to not explode.  I added `expect(snowplow).not.toHaveBeenCalled()`.